frontend/packages/knative-plugin/src/topology/components/nodes/EventSink.tsx

```diff
@@ -0,0 +1,158 @@
+  element,
+  selected,
+  onSelect,
+  onContextMenu,
+  contextMenuOpen,
+  dragNodeRef,
+  dndDropRef,
+  dragging,
+  edgeDragging,
+  onShowCreateConnector,
+  onHideCreateConnector,
+}) => {
+  const svgAnchorRef = useSvgAnchor();
+  const [hover, hoverRef] = useHover();
+  const groupRefs = useCombineRefs(dragNodeRef, dndDropRef, hoverRef);
+  const { data, resources, resource } = element.getData();
+  const [filtered] = useSearchFilter(element.getLabel(), resource?.metadata?.labels);
+  const displayFilters = useDisplayFilters();
+  const showLabelsFilter = getFilterById(SHOW_LABELS_FILTER_ID, displayFilters);
+  const showLabels = showLabelsFilter?.value || hover;
+  const { width, height } = element.getBounds();
+  const size = Math.min(width, height);
+  const allowEdgeCreation = useAllowEdgeCreation();
+  const isKafkaConnectionLinkPresent =
+    element.getSourceEdges()?.filter((edge: Edge) => edge.getType() === TYPE_KAFKA_CONNECTION_LINK)
+      .length > 0;
+  const revisionIds = resources.revisions.map((revision) => revision.metadata.uid);
+  const { loaded, loadError, pods } = usePodsForRevisions(revisionIds, resource.metadata.namespace);
+  const donutStatus = React.useMemo(() => {
+    if (loaded && !loadError) {
+      const [current, previous] = pods;
+      const isRollingOut = !!current && !!previous;
+      return {
+        obj: resource,
+        current,
+        previous,
+        isRollingOut,
+        pods: [...(current?.pods || []), ...(previous?.pods || [])],
+      };
+    }
+    return null;
+  }, [loaded, loadError, pods, resource]);
+
+  React.useLayoutEffect(() => {
+    if (allowEdgeCreation && !isKafkaConnectionLinkPresent) {
+      if (hover) {
+        onShowCreateConnector && onShowCreateConnector();
+      } else {
+        onHideCreateConnector && onHideCreateConnector();
+      }
+    }
+  }, [
+    hover,
+    onShowCreateConnector,
+    onHideCreateConnector,
+    allowEdgeCreation,
+    isKafkaConnectionLinkPresent,
+  ]);
+
+  return (
+    <g
+      className={classNames('odc-event-source', {
+        'is-filtered': filtered,
+        'is-dragging': dragging || edgeDragging,
+        'is-selected': selected,
+      })}
+      onClick={onSelect}
+      onContextMenu={onContextMenu}
+      ref={groupRefs}
+    >
+      <NodeShadows />
+      <polygon
+        key={hover || dragging || contextMenuOpen ? 'polygon-hover' : 'polygon'}
+        className="odc-event-source__bg"
+        ref={svgAnchorRef}
+        filter={createSvgIdUrl(
+          hover || dragging || contextMenuOpen
+            ? NODE_SHADOW_FILTER_ID_HOVER
+            : NODE_SHADOW_FILTER_ID,
+        )}
+        points={`${width / 2}, ${(height - size) / 2} ${width - (width - size) / 2},${height /
+          2} ${width / 2},${height - (height - size) / 2} ${(width - size) / 2},${height / 2}`}
+      />
+      {donutStatus && <PodSet size={size * 0.75} x={width / 2} y={height / 2} data={donutStatus} />}
```

Comment:
Looks like except the PodSet the most of other elements are same for Eventsource.tsx, if yes, can we abstract out a generic BasePolygonNode and make the code DRY WDYT?